File: gyozelem/Domain/FSObject.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain
{

    public enum FSFlag {
        Normal,
        Hidden,
        Protected
    }

    public enum FSStatus { 
        None        = 0, 
        Error       = 1,
        Uploading   = 2,
        Processing  = 3,
        Ok          = 4
    }

    public enum FSType {
        UNKNOWN    = 0,
        FOLDER     = 1,
        IMAGE      = 2,
        VIDEO      = 3,
        AUDIO      = 4,
        DOCUMENT   = 5,
        ARCHIVE    = 6,
        FONT       = 7,
    }

    public class FSMetaData
    {
        #nullable enable
        public string? Extension { get; set; }
        public string? Filename { get; set; }
        public string? MimeType { get; set; }
        #nullable disable
    }

    public static class FSObjectConfig {
        public const string BaseFolderName = "wwwroot";
        public const string AssetFolderName = "assets";
        public static string GetWholePath(string filename = null)
        {
            var path = FSObjectConfig.BaseFolderName + "/" + FSObjectConfig.AssetFolderName + "/";
            if (filename == null) {
                return path;
            }

            return path + filename;
        }
        public static string GetRelativePath(string filename)
        {
            return FSObjectConfig.AssetFolderName + "/" + filename;
        }
    }


    public class FSObject
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public Guid ParentId { get; set; }
        public FSStatus Status { get; set; }
        public FSType Type { get; set; }
        public int Flag { get; set; }
        public uint Size { get; set; }
        public string Extension { get; set; }
        public string Url { get; set; }
        public FSMetaData MetaData { get; set; }
        public DateTime CreatedAt { get; set; }
        public Guid CreatedBy { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public Guid? UpdatedBy { get; set; }
    }
}

File: gyozelem/client-app/src/components/file-explorer/model/FSObject.ts
import { BaseEntity, IBaseEntity } from "../../../core/model/BaseEntity";
import { CV } from "../../form-validator/validator";
// import { IBaseModel } from "../../form-validator/BaseModel";

export enum FSTypeEnum {
    UNKNOWN  = 0,
    FOLDER   = 1,
    IMAGE    = 2,
    VIDEO    = 3,
    AUDIO    = 4,
    DOCUMENT = 5,    
}

export enum FSStatusEnum {
    NONE = 0, 
    ERROR = 1,
    UPLOADING,
    PROCESSING,
    OK   
}

export const rootFolder = {
    id: '00000000-0000-0000-0000-000000000000',
    status: 1,
    name: 'root',
    parentId: null,
    size: 0,
    type: FSTypeEnum.FOLDER,
    url: '',
    createdAt: new Date(0),
    createdBy: '00000000-0000-0000-0000-000000000000'
} as IFSObject;

export interface IFSObject extends IBaseEntity<IFSObject> {
    status: number;
    name: string;
    parentId: string;
    type: FSTypeEnum;
    size: number;
    flag?: number;
    url?: string;
    extension?: string;
    metaData?: IFSOMetaData;
}

export interface IFSOMetaData {
    color?: string;
    duration?: number;
    extension?: string;
    height?: number;
    mime?: string;
    filename?: string;
    originalUrl?: string;
    size?: number;
    url?: string;
    width?: number;
}

export class FSObject extends BaseEntity<FSObject> implements IFSObject {

    public status: number;

    @CV('REQUIRED', 'Kell egy nev')
    public name: string;
    public parentId: string;
    public type: FSTypeEnum;
    public url?: string;
    public size: number;
    public flag?: number;
    public extension?: string;
    public metaData?: IFSOMetaData;
}

export default FSObject;
